### DISCUSSION TITLE
Questions about mechanoid features

### DISCUSSION THREAD

**Discussion by TAOFOR4** [hybridgroup/mechanoid#31] [General]
Hi, firstly, thx for this brilliant project. I just knew this project recently so have some questions:

1. Do you have a supported hardware list? I am doing research about wasm and tiny IoT devices. So I am wondering if you guys have a hardware list for mechanoid that I can test wasm with.
2. What is the advantage of mechanoid compared with other runtimes for wasm and tiny IoT devices, like wasm3 and wamr...
3. Will mechanoid support more languages in the future? I saw you support Rust which is very popular for IoT projects. But C is also a good candidate.

Thank you! :)

**Comment by orsinium** [comment]
> Do you have a supported hardware list?

Any platform supported by TinyGo. And that's quite a list!

https://tinygo.org/docs/reference/microcontrollers/

> What is the advantage of mechanoid compared with other runtimes for wasm and tiny IoT devices

The answer is in your previous question: crazy portability. Well, and Go is a friendlier and easier language than C. If you can sacrifice a bit of RAM for better DX and memory safety, Go is your best bet.

> like wasm3 

It's currently in low maintenance mode, see [their readme](https://github.com/wasm3/wasm3)

> Will mechanoid support more languages in the future?

The guest modules can be written in any language that compiles into wasm! [Rust](https://github.com/hybridgroup/mechanoid-templates/tree/main/modules/pingrs) is just an example. We also have examples for [Zig](https://github.com/hybridgroup/mechanoid-templates/tree/main/modules/pingzig) and [Go](https://github.com/hybridgroup/mechanoid-templates/tree/main/modules/ping). But there are many more languages that can compile into wasm. See [awesome-wasm-langs](https://github.com/appcypher/awesome-wasm-langs). Mechanoid doesn't need to do anything to support these languages. Just pick a language, compile into wasm similar to other examples we have, and all should work just fine.

**Reply by TAOFOR4** [reply]
Thanks for the detailed answer. Also, I found wazero, extism and other runtimes are also built on TinyGo. So compared with those, does mechanoid have any special features?

**Reply by orsinium** [reply]
> I found wazero

Mechanoid internally uses Wazero, so same thing. In fact, we are the people who added tinygo support in wazero to make it work with mechanoid.

> extism

Extism is in Rust.

> and other runtimes

There are no other maintained wasm runtimes in Go/TinyGo.
